### ISSUE TITLE
Problems with Vue & Webpack docs

### ISSUE THREAD

**Issue by herrbischoff** [pagekit/pagekit#720]
## Problem

I'm testing Pagekit for the first time and wanted to use Vue and Webpack, following the documentation at https://pagekit.com/docs/developer/vuejs-and-webpack to the letter. However, there are apparently problems with it and ultimately I wasn't able to get it to produce any output at all.
### Zero

I always get this console error, no matter what I do:

```
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (vue.min.js.map, line 0)
```
### One

```
<?php $view->script('example', 'hello:app/bundle/example', 'vue') ?>
```

This gets me a browser console error:

```
[Error] Failed to load resource: unsupported URL (hello:app/bundle/example, line 0)
```

I could fix that by adding the missing `.js` suffix and exchanging `hello` for `theme`.
### Two

Although only the one console error (see: Zero) appears, there is just no output. The rest of the page renders fine and simple inline JS like `console.log("It's alive!")` works.
## Technical Details
- Pagekit version: 1.0.5
- Webserver: Apache/2.4.16 (Ubuntu)
- Database: MySQL 5.5.46-0ubuntu0.14.04.2
- PHP Version: 5.6.14-1+deb.sury.org~trusty+1
- Browser: Safari Version 9.1.2 (11601.7.7)

**Comment by crwgregory** [comment]
Does webpack give you any errors? 

Can you confirm that webpack is creating the files in your bundle directory?

**Comment by herrbischoff** [comment]
Yes, indeed. Webpack is working correctly and creates the files as expected. Otherwise I would have seen 404 errors. Also, I have checked the source of the created files — they all seem to be in order.

**Comment by crwgregory** [comment]
So it's a problem with requested the resource.

Try passing 'vue' in an array, might work.

`<?php $view->script('example', 'hello:app/bundle/example.js', ['vue']) ?>`

> `[Error] Failed to load resource: unsupported URL (hello:app/bundle/example, line 0)`

Tells me that the 'hello' module isn't being registered properly?

[You can use this template generator to make a clean extension to see if that works for you too.](https://github.com/drunken-kungfu/pagekit-template-generator)

**Comment by MalteScharenberg** [comment]
@herrbischoff This looks like the asset manger is not able to locate your file. Whats the name of your extension? I think if it is not `hello` you have to register the `hello:` resource locater within the module definition:

```
 'resources' => [
        'hello:' => ''
    ]
```

Or change the resource locator accordantly when requiring assets. 

@crwgregory For one dependency it's okay to pass a string. :)

**Comment by crwgregory** [comment]
I've ran into other people on Gitter with who are very confused when they try to set up Pagekit to use Webpack. I've looked over the docs and they could use some work. A couple of points I'd like to bring up:

- The docs say to use Webpack from the base of pagekit, however when one tries to do that, you get an error that the module 'lodash' cannot be found. If you run npm install, it breaks more things then it fixes.
- I would suggest changing the docs to explain how to create a local `package.json` file that the user can use to npm install in their project and changing the docs to tell the user to use Webpack at their package root and not Pagekits root. 
- The docs explain how to compile the component and import the file *before* explaining how to register it either as a global component, or a component in a regular .js file that uses the `Vue.ready` function and binds to an `el`. I don't know of a way you can use a component without registering it some how first. You can't just Webpack a Vue file with only a `module.exports = {};` and expect to be able to use that component by just importing the js file. If I am wrong please correct me.

I started going over the docs and making some changes but quickly realized that the changes that I think need to be made are kind of extensive so I wanted to run it by the team.

**Comment by florianletsch** [comment]
This is a good suggestion. If you would like to submit changes to https://github.com/pagekit/docs, that would be highly appreciated @crwgregory

**Comment by crwgregory** [comment]
@florianletsch I haven't had allot of time to get to this and honestly don't know when I would be able to. Just letting you know so you are not waiting on me... I will advise if I do start work on it.

**Comment by herrbischoff** [comment]
As nothing is moving for several months now, I'm going to close this issue. I have dropped working with PageKit in the meantime as it's sadly not ready for primetime yet. Also, another issue regarding RSS was completely ignored, so that's that.
